dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/patch/DefaultPatchService.java

```diff
@@ -195,11 +196,13 @@ private Patch diff( JsonNode jsonNode )
         {
             Collection addCollection = ReflectionUtils.newCollectionInstance( property.getKlass() );
 
-            Collection sourceCollection = (Collection) ((Collection) sourceValue).stream()
-                .map( o -> ((IdentifiableObject) o).getUid() ).collect( Collectors.toList() );
+            Collection sourceCollection = (Collection) ( (Collection) sourceValue ).stream()
+                .filter( Objects::nonNull )
+                .map( o -> ( (IdentifiableObject) o ).getUid() ).collect( Collectors.toList() );
 
```

Comment:
got NullPointerException here, so I need to filter null item from the sourceValue list